Automattic/woocommerce-payments
# Adapt single currency settings page UI to match WCPay theme

Fixes #2874

#### Changes proposed in this Pull Request


Create a React rendered component for single currency settings


This PR provides an UI for the single currency settings compatible with the new WCPay grouped settings interface, which enables users see the current rate of a currency, when it's last updated, set manual rates for them, set price rounding fraction, set price charm options, and test their settings instantly with a preview section, which has the field for setting a base amount and see the converted amount regarding the new settings instantly.

The new screen looks like this:

![image](https://user-images.githubusercontent.com/3295/133225853-84a20f05-bd31-4c84-9ce1-ba08a62bc9fb.png)

One note; Since the main currency settings will be updated on #2832, the header part couldn't be removed yet, it'll be done when merging this with #2832 to the `develop` branch.

And one nitpick; I needed to use a context provider for both currency selection screen and single currency screen to exchange display statuses and open/close screen methods like this:

```jsx
<MultiCurrencySettingsContext.Provider
    value={ {
        isSingleCurrencyScreenOpen: isSingleCurrencyScreenOpen,
        currencyCodeToShowSettingsFor: currencyCodeToShowSettingsFor,
        openSingleCurrencySettings: handleOpenSingleCurrencySettings,
        closeSingleCurrencySettings: handleCloseSingleCurrencySettings,
    } }
>
    { ! isSingleCurrencyScreenOpen ? (
        <EnabledCurrencies />
    ) : (
        <SingleCurrencySettings />
    ) }
</MultiCurrencySettingsContext.Provider>
```

Taking this into account, it can be easily merged with #2832 and be ready to use.





#### Testing instructions

##### Automatic testing

- Run `npm run test:js single-currency-settings``

##### Manual testing

- Open *WooCommerce > Settings* and click to *Multi-Currency* tab
- Add some currencies to the list if the list is empty
- Click *Manage* on any currency
- See if the default settings show as 'Automatic' as the exchange rate selection, 'None' for the price charm, and '1.00' as the rounding mode for non-zero-decimal currencies, and '100' for the zero-decimal ones.
- Change some settings and save the changes
- Go back to the main screen with the **Enabled Currencies** link on the breadcrumb (I offered to add a back button to the screen), and @LevinMedia told me that it's an open issue for the Wizard component which is waiting on hold for now, and there's no need to add a separate one for this yet.)
- Open another currency settings, and check if the settings are correct, not displaying the previous one's settings
- Refresh the page and check if the previously set settings are correctly displayed
- Change the settings and see if the Preview div is updated correctly for the given settings





-------------------


**Post merge**
